worker/codalabworker/file_util.py

```diff
@@ -198,9 +200,12 @@ def ensure_ends_with_newline(lines, remove_line_without_newline=False):
             if num_head_lines > 0:
                 # To ensure that the last line is a whole line, we remove the
                 # last line if it doesn't have a newline character.
-                head_lines = fileobj.read(num_head_lines * max_line_length).splitlines(True)[
-                    :num_head_lines
-                ]
+                try:
+                    head_lines = fileobj.read(num_head_lines * max_line_length).splitlines(True)[
+                        :num_head_lines
+                    ]
+                except UnicodeDecodeError:
```

Comment:
I'm guessing that this exception is triggered whenever fileobj is binary. Why is that the case? I don't see anything specific for expecting string output in the line above.

Reply:
because the file is opened as a text file